SimTKcommon/tests/CMakeLists.txt

```diff
@@ -26,27 +26,54 @@ foreach(TEST_PROG ${REGR_TESTS})
     get_filename_component(TEST_ROOT ${TEST_PROG} NAME_WE)
 
     if(BUILD_TESTS_AND_EXAMPLES_SHARED)
-        # Link with shared library
-        add_executable(${TEST_ROOT} ${TEST_PROG})
-        set_target_properties(${TEST_ROOT}
-        PROPERTIES
-          PROJECT_LABEL "Test_Regr - ${TEST_ROOT}")
-        target_link_libraries(${TEST_ROOT} 
-                    ${TEST_SHARED_TARGET})
-        add_test(${TEST_ROOT} ${EXECUTABLE_OUTPUT_PATH}/${TEST_ROOT})
+        # TestAutodiffADOLCCommon was designed to test the ADOL-C libraries only
+        # and is therefore excluded from the tests of the shared libraries.
+        # TODO need a better way to link tests to ADOL-C libraries.
```

Comment:
One option is to have separate GLOBs for ADOLC vs non-ADOLC tests. The structure of this file could be as follows:

```
file(GLOB ALL_REGR_TESTS "*.cpp")
set(REGR_TESTS ${ALL_REGR_TESTS})
file(GLOB ADOLC_SPECIFIC_TESTS "*TestAutodiffADOLCCommon*")
list(REMOVE_ITEM REGR_TESTS ${ADOLC_SPECIFIC_TESTS})

foreach(TEST_PROG ${REGR_TESTS})
    # leave this loop as it used to be.
endforeach()

if(BUILD_ADOLC_LIBRARIES)
    set(ADOLC_REGR_TESTS ${ALL_REGR_TESTS})
    file(GLOB ADOLC_TESTS_TO_EXCLUDE "*TestXml*")
    list(REMOVE_ITEM ADOLC_REGR_TESTS ${ADOLC_TESTS_TO_EXCLUDE})
    foreach(TEST_PROG ${ADOLC_REGR_TESTS})
        # create the ADOLC tests.
    endforeach()
endif()

I am indifferent as to whether this is done in this PR or in another PR later.
```